File: backend/apps/hostel/models/hostel.py
from django.db import models


class Hostel(models.Model):
    """Maps to `hostel` in db_current."""

    id = models.AutoField(primary_key=True)
    hostel_name = models.CharField(max_length=100, blank=True, null=True)
    type = models.CharField(max_length=50, blank=True, null=True)
    address = models.TextField(blank=True, null=True)
    intake = models.IntegerField(blank=True, null=True)
    description = models.TextField(blank=True, null=True)
    hostel_incharge = models.CharField(max_length=44)
    is_active = models.CharField(max_length=255, blank=True, null=True, default='no')
    created_at = models.DateTimeField()
    updated_at = models.DateField(blank=True, null=True)
    meal_type = models.TextField(blank=True, null=True)

    class Meta:
        managed = False
        db_table = "hostel"

    def __str__(self):
        return f"Hostel {self.pk}"

File: frontend/src/types/hostel/hostel.ts
import type { ActiveFlag } from '@app-types/settings/session';

export interface Hostel {
  id: number;
  hostel_name: string | null;
  type: string | null;
  address: string | null;
  intake: number | null;
  description: string | null;
  hostel_incharge: string | null;
  is_active: ActiveFlag | string | null;
  meal_type: string | null;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface CreateHostelPayload {
  hostel_name: string;
  type?: string | null;
  address?: string | null;
  intake?: number | null;
  description?: string | null;
  hostel_incharge?: string | null;
  is_active?: string;
  meal_type?: string | null;
}

export type UpdateHostelPayload = Partial<CreateHostelPayload>;
